modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaSttpClientCodegen.java

```diff
@@ -87,4 +207,290 @@ public CodegenOperation fromOperation(String path,
+        }
+
+        @Override
+        public CliOption toCliOption() {
+            return CliOption.newString(name, description).defaultValue(defaultValue);
+        }
+
+        @Override
+        public void updateAdditionalProperties(Map<String, Object> additionalProperties) {
+            if (!additionalProperties.containsKey(name)) {
+                additionalProperties.put(name, defaultValue);
+            }
+        }
+
+        @Override
+        public String getValue(Map<String, Object> additionalProperties) {
+            return additionalProperties.getOrDefault(name, defaultValue).toString();
+        }
+    }
+
+    private static class BooleanProperty extends Property<Boolean> {
+        private BooleanProperty(String name, String description, Boolean defaultValue) {
+            super(name, description, defaultValue);
+        }
+
+        @Override
+        public CliOption toCliOption() {
+            return CliOption.newBoolean(name, description, defaultValue);
+        }
+
+        @Override
+        public void updateAdditionalProperties(Map<String, Object> additionalProperties) {
+            Boolean value = getValue(additionalProperties);
+            additionalProperties.put(name, value);
+        }
+
+        @Override
+        public Boolean getValue(Map<String, Object> additionalProperties) {
+            return Boolean.valueOf(additionalProperties.getOrDefault(name, defaultValue.toString()).toString());
+        }
+    }
+
+    private static class JsonLibraryProperty extends StringProperty {
+        private static final String JSON4S = "json4s";
+        private static final String CIRCE = "circe";
+
+        private JsonLibraryProperty() {
+            super("jsonLibrary", "Json library to use. Possible values are: json4s and circe.", JSON4S);
+        }
+
+        @Override
+        public void updateAdditionalProperties(Map<String, Object> additionalProperties) {
+            String value = getValue(additionalProperties);
+            if (value.equals(CIRCE) || value.equals(JSON4S)) {
+                additionalProperties.put(CIRCE, value.equals(CIRCE));
+                additionalProperties.put(JSON4S, value.equals(JSON4S));
+            } else {
+                IllegalArgumentException exception =
+                        new IllegalArgumentException("Invalid json library: " + value + ". Must be " + CIRCE + " " +
+                                "or " + JSON4S);
+                throw exception;
+            }
+        }
+    }
+
+    private static class MainPackageProperty extends StringProperty {
+
+        private MainPackageProperty() {
+            super("mainPackage", "Top-level package name, which defines 'apiPackage', 'modelPackage', " +
+                    "'invokerPackage'", DEFAULT_PACKAGE_NAME);
+        }
+
+        @Override
+        public void updateAdditionalProperties(Map<String, Object> additionalProperties) {
+            if (additionalProperties.containsKey(name)) {
```

Comment:
Hmm, this is definitively a good idea to cover it with some tests

Reply:
It isn't an error since the `mainPackage` property isn't used anywhere. All packages are specified by `apiPackage`, `modelPackage` and `invokerPackage` properties